core/services/ocr2/plugins/ocr2keeper/evm/offchain_lookup.go

```diff
@@ -0,0 +1,349 @@
+	offchainLookup.callbackFunction = *abi.ConvertType(errorParameters[3], new([4]byte)).(*[4]byte)
+	return offchainLookup, nil
+}
+
+// offchainLookupCallback calls the callback(bytes calldata extraData, string[] calldata values, uint256 statusCode) specified by the
+// 4-byte selector from the revert. the return will match check telling us if the upkeep is needed and what the perform data is
+func (r *EvmRegistry) offchainLookupCallback(ctx context.Context, offchainLookup OffchainLookup, values []string, statusCode int, upkeepInfo keeper_registry_wrapper2_0.UpkeepInfo, opts *bind.CallOpts) (bool, []byte, error) {
+	// call to the contract function specified by the 4-byte selector callbackFunction
+	typBytes, err := abi.NewType("bytes", "", nil)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "abi new bytes type error:")
+	}
+	typStrings, err := abi.NewType("string[]", "", nil)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "abi new strings type error:")
+	}
+	typUint, err := abi.NewType("uint256", "", nil)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "abi new uint256 type error:")
+	}
+	callbackArgs := abi.Arguments{
+		{Name: "extraData", Type: typBytes},
+		{Name: "values", Type: typStrings},
+		{Name: "statusCode", Type: typUint},
+	}
+	pack, err := callbackArgs.Pack(offchainLookup.extraData, values, big.NewInt(int64(statusCode)))
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "callback args pack error:")
+	}
+
+	var callbackPayload []byte
+	callbackPayload = append(callbackPayload, offchainLookup.callbackFunction[:]...)
+	callbackPayload = append(callbackPayload, pack...)
+
+	checkUpkeepGasLimit := uint32(200000) + uint32(6500000) + uint32(300000) + upkeepInfo.ExecuteGas
+	callbackMsg := ethereum.CallMsg{
+		From: r.addr,             // registry addr
+		To:   &upkeepInfo.Target, // upkeep addr
+		Gas:  uint64(checkUpkeepGasLimit),
+		Data: hexutil.Bytes(callbackPayload), // function callback(bytes calldata extraData, string[] calldata values, uint256 statusCode)
+	}
+
+	callbackResp, err := r.client.CallContract(ctx, callbackMsg, opts.BlockNumber)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "call contract callback error:")
+	}
+
+	boolTyp, err := abi.NewType("bool", "", nil)
+	callbackOutput := abi.Arguments{
+		{Name: "upkeepNeeded", Type: boolTyp},
+		{Name: "performData", Type: typBytes},
+	}
+	unpack, err := callbackOutput.Unpack(callbackResp)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "callback ouput unpack error:")
+	}
+
+	upkeepNeeded := *abi.ConvertType(unpack[0], new(bool)).(*bool)
+	if !upkeepNeeded {
+		return false, nil, nil
+	}
+	performData := *abi.ConvertType(unpack[1], new([]byte)).(*[]byte)
+	return true, performData, nil
+}
+
+func (r *EvmRegistry) doRequest(o OffchainLookup, upkeepId *big.Int, offchainConfig []byte) ([]byte, int, error) {
+	client := http.Client{}
```

Comment:
Should you specify a timeout on the client (I think by default there is no timeout)?